skyportal/models.py

```diff
@@ -463,15 +462,16 @@ def last_detected(cls):
     def add_linked_thumbnails(self):
         """Determine the URLs of the SDSS and DESI DR8 thumbnails of the object,
         insert them into the Thumbnails table, and link them to the object."""
-        sdss_thumb = Thumbnail(
-            photometry=self.photometry[0], public_url=self.sdss_url, type='sdss'
-        )
-        dr8_thumb = Thumbnail(
-            photometry=self.photometry[0], public_url=self.desi_dr8_url, type='dr8'
-        )
+        sdss_thumb = Thumbnail(obj=self, public_url=self.sdss_url, type='sdss')
+        dr8_thumb = Thumbnail(obj=self, public_url=self.desi_dr8_url, type='dr8')
         DBSession().add_all([sdss_thumb, dr8_thumb])
         DBSession().commit()
 
+    def add_ps1_thumbnail(self):
+        ps1_thumb = Thumbnail(obj=self, public_url=self.panstarrs_url, type="ps1")
+        DBSession().add(ps1_thumb)
+        DBSession().commit()
+
```

Comment:
Why have these separate?

Reply:
Because PS1 is the only one that makes an external request to generate the link - the other two just immediately return a string, so are super quick and can be added as soon as the obj is created without much of a performance hit